src/App.js

```diff
@@ -129,16 +143,22 @@ function App() {
   const [giveup,setGiveup] = useState(false);
   const inputRef = useRef();
 
-  const matches = useMemo(() => {
+  const {matches, newMatches, newMisMatches} = useMemo(() => {
     const nextMatches = {};
-    guesses.forEach((guess) => {
+    let newMatch = 0, newMisMatch = 0;
+    guesses.forEach((guess, guessIndex) => {
+      const isLastGuess = guessIndex === guesses.length - 1;
       indexes.forEach((index) => {
         const result = check(index, guess, word);
-        if (nextMatches[guess[index]] !== "match")
+        if (nextMatches[guess[index]] !== "match") {
           nextMatches[guess[index]] = result;
+          if (result === "match" && isLastGuess) newMatch++;
+          if (result === "mismatch" && isLastGuess) newMisMatch++;
+        };
       });
     });
-    return nextMatches;
+    console.log(newMatch, newMisMatch);
+    return {matches: nextMatches, newMatches: newMatch, newMisMatches: newMisMatch};
```

Comment:
Rather than have to rename "newMatch" to "newMatches", I'd recommend calling the variables "newMatches" and "newMisMatches" throughout, starting at line 148 where they are declared.